Src/ILGPU/Context.cs

```diff
@@ -384,6 +385,70 @@ public DeviceCollection<TDevice> GetDevices<TDevice>()
                 : new DeviceCollection<TDevice>(new List<Device>());
         }
 
+        /// <summary>
+        /// Attempts to return the most optimal single device.
+        /// </summary>
+        /// <param name="preferCPU">Always returns CPU device 0.</param>
+        /// <returns>Selected device.</returns>
+        public Device GetPreferredDevice(bool preferCPU)
```

Comment:
SUGGESTION: Replace the implementation of this function, and call `GetPreferredDevices` instead.

```CSharp
public Device GetPreferredDevice(bool preferCPU) =>
    GetPreferredDevices(preferCPU, matchingDevicesOnly: false).First();
```

Less code to maintain.